go/libraries/doltcore/sqle/database.go

```diff
@@ -1574,10 +1566,15 @@ func (db Database) GetViewDefinition(ctx *sql.Context, viewName string) (sql.Vie
 		return sql.ViewDefinition{Name: viewName, TextDefinition: blameViewTextDef, CreateViewStatement: fmt.Sprintf("CREATE VIEW `%s` AS %s", viewName, blameViewTextDef)}, true, nil
 	}
 
-	key, err := doltdb.NewDataCacheKey(root)
+	schTblHash, ok, err := root.GetTableHash(ctx, doltdb.TableName{Name: doltdb.SchemasTableName})
```

Comment:
Need to take the schema name from `db` (or pass it), otherwise this will false-positive for different schemas in doltgres